app/src/main/java/com/portfolio/tasky/entry/data/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,30 @@
+package com.portfolio.tasky.entry.data
+
+import android.content.Context
+import com.google.gson.Gson
+import com.portfolio.tasky.R
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import javax.inject.Inject
+
+class UserPreferencesImpl @Inject constructor(val context: Context) : UserPreferences {
+
+    private val sharedPreference by lazy { context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)}
+
+    override fun saveAuthenticatedUser(authenticatedUser: AuthenticationResponse?) {
```

Comment:
Make it non nullable

What would be the use of saving a null? None (or at least, I can't think of one)
So no need to do it

Reply:
Agreed